### ISSUE TITLE
`used-before-assignment` false positive on multiple-target assignment

### ISSUE THREAD

**Issue by xmo-odoo** [PyCQA/pylint#5955]
### Bug description

When assigning to multiple target, and target `n` is used in target `n+1`, pylint flags the the line as E0601 "used-before-assignment", despite the statement being entirely correct per Python semantics:
```python
# pylint: disable=missing-module-docstring, invalid-name
b = {}
a = b[id(a)] = 0
```
Even if there's an argument for flagging it (it's not great style), I don't think E0601 is valid.

### Command used

```shell
`pylint <file>.py`
```

### Pylint output

```shell
************* Module <file>
<file>.py:3:9: E0601: Using variable 'a' before assignment (used-before-assignment)

-----------------------------------------------------------------------
Your code has been rated at -15.00/10 (previous run: -25.00/10, +10.00)
```

### Expected behavior

That the code passes,  possibly triggers some sort of code style warning at most.

### Pylint version

```shell
pylint 2.12.2
astroid 2.9.0
Python 3.8.12 (default, Sep 27 2021, 09:39:37)
```

**Comment by tushar-deepsource** [comment]
Similar case:

```python
x = [10]
y = 20
z, z[0] = x.copy(), y
```

**Comment by tushar-deepsource** [comment]
Do you think it's sufficient to suppress it if that variable is present before the current invocation in the assign targets in that same assign statements?

**Comment by Pierre-Sassoulas** [comment]
Could you provide an example @tushar-deepsource ?

**Comment by tusharsadhwani** [comment]
@Pierre-Sassoulas

where we are doing `self.add_message("used-before-assignment")`, if we checked that there exists an `AssignName` node for the same variable in the same statement before it, and only raise it if there isn't one.

**Comment by Pierre-Sassoulas** [comment]
I meant a python example of what should raise or not :) (something that could become a functional tests if you will)

**Comment by tusharsadhwani** [comment]
@Pierre-Sassoulas

```python
items = []
x = y = items[x] = 0  # Don't raise here

a, items[a] = 0, 20  # Don't raise here

print(b, c)  # Raise for `b` here

b = 10
````\
